### ISSUE TITLE
0 bytes files get ignored except for the first

### ISSUE THREAD

**Issue by alon** [jxxcarlson/elm-tar#1]
Sorry for the not complete bug report - I just wanted to report early and fill in the example later.

I tried to compress 3 files each with a unique name and zero length'ed content, the resulting tar file contained just one of the files. When I changed the content to be not empty the result was correct, i.e. all three files.

I will fill this in with an actual example later.

**Comment by alon** [comment]
Something like:

createArchive [
  ({ defaultFileRecord | filename = "test1.csv"}, StringData ""),
  ({ defaultFileRecord | filename = "test2.csv"}, StringData "")
]

which results in a tar file with only one of the files

**Comment by jxxcarlson** [comment]
Thanks very much @alon, I am looking into this now.

**Comment by jxxcarlson** [comment]
@alon, I pushed version 3.0.8 of `elm-tar` to the package manager, which may provide an interim fix for your problem.  Please let me know what happens.  I ran your example successfully, but with a name change to accommodate some changers in the API (**defaultMetada**):
:
```
arx =
    createArchive
        [ ( { defaultMetadata | filename = "test1.csv" }, StringData "" )
        , ( { defaultMetadata | filename = "test2.csv" }, StringData "" )
        ]
```
The problem you were experiencing was  due to the fact  that I was putting a 512 byte block of zeros for the data block in the case of an empty string. That block should be nonexistent in the latter case.  Also, please note the following file types (or not):

## Metadata notes

There is an unresolved problem of how to properly encode the metadata
of text files so that `extractArchive` knows to decode the content
as a string.  At the moment, archived files whose content is to be
decoded as a string are recognized by their file extension.  The
admissible text files have extension  in the list

```
[ "text", "txt", "tex", "csv" ]
```

This is an unsatisfactory solution, but at the moment, I don't
know how to get around it.
then 

```
> arx
<3072 bytes> : Bytes.Bytes

> exx = extractArchive arx
[({ fileNamePrefix = "", fileSize = 0, filename = "test1.csv", groupID = 123, groupName = "", lastModificationTime = 0, linkIndicator = NormalFile, linkedFileName = "foo.txt", mode = { group = [Write,Read], other = [Read], system = [], user = [Execute,Write,Read] }, ownerID = 501, userName = "" },StringData ""),({ fileNamePrefix = "", fileSize = 0, filename = "test2.csv", groupID = 123, groupName = "", lastModificationTime = 0, linkIndicator = NormalFile, linkedFileName = "foo.txt", mode = { group = [Write,Read], other = [Read], system = [], user = [Execute,Write,Read] }, ownerID = 501, userName = "" },StringData "")]

> List.length exx
2 : Int
```

**Comment by jxxcarlson** [comment]
Fixed by e238e83edac68679e5a9d048bee5fcde86db43fb.  ADDED: `Test.md`  --- explanatiion of the tests.
